spec/models/async_indexer_spec.rb

```diff
@@ -42,4 +42,44 @@
     AsyncIndexer.perform("WorkIndexer:34:#{Time.now.to_i}")
   end
 
+  context "when persistent failures occur" do
+    before do
+      # Make elasticsearch always fail.
+      allow($new_elasticsearch).to receive(:bulk) do |options|
+        {
+          "errors" => true,
```

Comment:
👍

For the record, this is the error output to fake: https://www.elastic.co/guide/en/elasticsearch/reference/6.0/docs-bulk.html